janMarkusMilan/assignments/treesBFS/lc107/lc107.js

```diff
@@ -1,29 +1,140 @@
 // Problem Statement #
 
-// Given a binary tree, populate an array to represent its level-by-level traversal in reverse order, i.e., the lowest level comes first. You should populate the values of all nodes in each level from left to right in separate sub-arrays.
+// Given a binary tree, populate an array to represent its level-by-level traversal in reverse order, i.e., the lowest level comes first.
+// You should populate the values of all nodes in each level from left to right in separate sub-arrays.
 
-class TreeNode {
+/*
+Format: [[1], [2, 3], [4, 5, 6, 7], ...]
+Edge Case: !root -> return []
+
+BFS: Level-by-level traversal
+
+function BFS (root) {
+  res = []
+  queue = [root]
+
+  while (queue.length) {
+    node = queue.shift
+    res.push(node.val)
+
+    if (node.left) {
+      queue.push(node.left)
+    }
+
+    if (node.right) {
+      queue.push(node.right)
+    }
+  }
+
+  return res;
+}
 
+BFS: [1, 2, 3, 4, 5, 6, ...]
+
+Modify the pseudo code for BFS:
+  1. Instead of top-down, we go bottom-up
+  2. Keep track of the length of the levels (var = queue.length)
+
+Logic (main modifications):
+  1. Creating a temp array to store the values in each level
+  2. Loop -> condition: until we reach the end of the level
+  3. Push the values of the node in our array in each iteration
+  4. A counter -> once it reaches the desired length (end of level), push our sub-array into our res array
+*/
```

Comment:
Great approach. It may help your interviewer if instead of writing the BFS code you write what the code will do in sentences.